isis/src/base/objs/Cube/Cube.cpp

```diff
@@ -507,7 +508,52 @@ namespace Isis {
       core.addGroup(ptype);
     }
     else if (labelsAttached() == LabelAttachment::ExternalLabel) {
-      cubFile = cubFile.addExtension("ecub");
+      if (!m_dataFileName) {
+        if (format() == Bsq || format() == Tile) {
+          cubFile = cubFile.addExtension("cub");
+
+          Pvl dnLabel;
+          PvlObject isiscube("IsisCube");
+          PvlObject dnCore(core);
+
+          dnCore.addGroup(dims);
+          dnCore.addGroup(ptype);
+
+          isiscube.addObject(dnCore);
+          dnLabel.addObject(isiscube);
+
+          Cube dnCube;
+          dnCube.fromLabel(cubFile, dnLabel, "rw");
+          dnCube.close();
+
+          m_dataFileName = new FileName(cubFile);
+          m_dataFile = new QFile(realDataFileName().expanded());
+        }
+        else if (format() == GTiff) {
+          GDALAllRegister();
+          GDALDriver* driver = GetGDALDriverManager()->GetDriverByName("GTiff");
+          if (driver) {
+            cubFile = cubFile.addExtension("tiff");
+            m_dataFileName = new FileName(cubFile);
+            std::string dataFileName = m_dataFileName->expanded().toStdString();
+            char **papszOptions = NULL;
+            GDALDataset *dataset = driver->Create(dataFileName.c_str(), sampleCount(), lineCount(), bandCount(), IsisPixelToGdal(pixelType()), papszOptions);
+            for (int i = 1; i <= bandCount(); i++) {
+              GDALRasterBand *band = dataset->GetRasterBand(i);
+              band->SetScale(multiplier());
+              band->SetOffset(base());
+            }
```

Comment:
We should make this more generic as a gdal type but is good to get things going

Reply:
I agree we should make this more generic but I think having the actual format is useful rather than just using `gdal` as a format. Generally we still need to know what kind of file we are going to be using for DNs.

I think adding something like `gdal+GTiff` as cube atts would be decent but from using this feature as is. There wouldn't be much of a difference from `gdal+GTiff` vs `external+GTiff`